api/api.js

```diff
@@ -1,93 +1,76 @@
 const express=require('express');
 const app=express();
+const MongoClient = require('mongodb').MongoClient;
+const url = 'mongodb://root:example@localhost:27017';
 const bodyParser=require('body-parser');
-const {mongoose} = require('./config/mongoose');
+const ObjectID = require('mongodb').ObjectID;
 
 app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
 
-//Mongoose models
-const { Cat } = require('./models/cat.model');
-const { Task } = require('./models/tasks.model');
- 
-//returns array of categorys
-app.get('/db/task-service/categories',(req,res)=>{
-    Cat.find({}).then((categorys)=>{
-        res.send(categorys);
+MongoClient.connect(url, { useNewUrlParser: true, useUnifiedTopology: true }, function (err, client) {
+    if (err) throw err;
+    
+    process.on('unhandledRejection', (reason, promise) => {
+      console.log('Unhandled Rejection at:', promise, 'reason:', reason);
+      // Application specific logging, throwing an error, or other logic here
     });
-})
-
-//adds a new category
-app.post('/db/task-service/categories',(req,res)=>{
-    let title = req.body.title;
-
-    let newCat = new Cat({
-        title
-    });
-    newCat.save().then((catDoc)=>{
-        res.send(catDoc);
+    app.post('/db/task-service/tasks', (req,res,next)=>{
+        var task={
+            _id: req.body.ObjectID,
+            text: req.body.text,
+            completed: req.body.completed
+        };
+        client.db('task-service').collection("tasks").insertOne(task, (err, result) =>{
```

Comment:
A try catch pattern can also be applied here. You will see this pattern in our card-service.

Reply:
I did this